### ISSUE TITLE
MySQL - `CREATE TABLE foo SELECT * FROM bar` statement causes `Found unparsable section` error 

### ISSUE THREAD

**Issue by gdubicki** [sqlfluff/sqlfluff#5852]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I am trying to validate some SQL scripts that create syntax like:

```sql
CREATE TABLE foo SELECT * FROM bar;
```

### Expected Behaviour

This syntax should work as [MySQL supports it](https://dev.mysql.com/doc/refman/5.7/en/create-table-select.html).

### Observed Behaviour

```
$ sqlfluff lint --dialect mysql my_script.sql 
== [my_script.sql] FAIL                                                                                                                                          
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE foo SELECT * FROM bar;'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

1. Create a `my_script.sql` file:
```sql
CREATE TABLE foo SELECT * FROM bar;
```
2. Try to lint it with:
```bash
sqlfluff lint --dialect mysql my_script.sql 
```

### Dialect

MySQL

### Version

```
$ sqlfluff --version
sqlfluff, version 3.0.5
```
...installed using pipx, Python 3.11.8.

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by MarkMarine** [comment]
I'm seeing the same parsing error when using CREATE FUNCTION ...

**Comment by KulykDmytro** [comment]
same in `ansi` and `athena`
